src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -29,6 +29,18 @@ public static bool IsWordCharacter(char ch)
         public static bool IsCompletionItemStartCharacter(char ch)
             => ch == '@' || IsWordCharacter(ch);
 
+        public static bool TreatAsDot(SyntaxToken token, int characterPosition)
+        {
+            if (token.Kind() == SyntaxKind.DotToken)
+                return true;
+
+            // if we're right after the first dot in .. then that's considered completion on dot.
+            if (token.Kind() == SyntaxKind.DotDotToken && token.SpanStart == characterPosition)
+                return true;
+
+            return false;
+        }
```

Comment:
this was a move.  it is used in SymbolCompletionProvider and UnnamedSymbolCompletionProvider.

Reply:
Just curious, why dot dot is also considered? (Sounds like  we are allowing the user to have a second chance to trigger this? )

Reply:
doc'ed in the helper.  but it's because users can have something like `foo.ToString()`.  they then type `.` after `foo` to get completion (perhaps to complete to a property).  That makes the code `foo..ToString()`.  The parser parses the `..` as a single token.  So we need to handle that specially to ensure we still show completion and do not think of that as a range expression.